### PR TITLE
Lig 1593 part2

### PR DESCRIPTION
**Pull request by japrescott**
fixes lig-1593
- ensure to only set s3 headers for server side encryption if we are actually talking to s3 as google buckets doesn't like headers of other buckets

closes lig-1596
- add docs about s3 server side encryption

### CODE REVIEW

File: lightly/api/utils.py [hunk 1]
```diff
@@ -130,4 +131,29 @@ def build_azure_signed_url_write_headers(content_length: str,
...
+    return headers
+
+
+class DatasourceType(Enum):
+    S3 = "S3"
+    GCS = "GCS"
+    AZURE = "AZURE"
+    LOCAL = "LOCAL"
+
+def get_signed_url_destination(signed_url: str)->DatasourceType:
```
Comment (MalteEbner) [inline lightly/api/utils.py:143]: I would write a unittest for this.

**Review by MalteEbner — CHANGES_REQUESTED** [review]
In general, this makes some nice improvements! You can resolve my appreciations if needed.

However, I would rework and unittest the `get_signed_url_destination` function.

File: lightly/api/utils.py [hunk 2]
```diff
@@ -130,4 +131,29 @@ def build_azure_signed_url_write_headers(content_length: str,
...
+    """
+    Tries to figure out the of which cloud provider/datasource type a signed url comes from (S3, GCS, Azure)
+    Args:
+        signed_url:
+            The signed url of a "bucket" provider
+    Returns:
+        DatasourceType
+    """
+    if signed_url.find('storage.googleapis.com/')!=-1 and  signed_url.find('boris-platform-')!=-1:
```
[the remark this answers is not in the document]
Comment (japrescott) [reply lightly/api/utils.py:152]: good point it was a mindless copy+paste from the UI

[the remark this answers is not in the document]
Comment (japrescott) [reply lightly/api/utils.py:152]: hmmkey but its not nullsafe. I will default the param

File: lightly/api/utils.py (hunk 1 above)
Comment (japrescott) [reply lightly/api/utils.py]: done

**Review by MalteEbner — APPROVED** [review]

**Review by MalteEbner — APPROVED** [review]
Great :)

**Pull request merged**